src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -1304,4 +1313,79 @@ public override Task<ImplementationResult> ImplementationAsync(DelegatedPosition
     }
 
     private record struct DelegationRequestDetails(string LanguageServerName, Uri ProjectedUri, ITextBuffer TextBuffer);
+
+    private class TelemetryReportingLSPRequestInvoker : LSPRequestInvoker
```

Comment:
I think this is worth being a class on its own, in its own file, and not nested. There are a few more spots that use `LSPRequestInvoker` and we might want to wrap them too eventually.